gotpm/cmd/pubkey.go

```diff
@@ -68,27 +68,11 @@ NVDATA instead (and --algo is ignored).`,
 	},
 }
 
-var keyAlgo string
-
 func init() {
 	RootCmd.AddCommand(pubkeyCmd)
 	addIndexFlag(pubkeyCmd)
 	addOutputFlag(pubkeyCmd)
-	pubkeyCmd.PersistentFlags().StringVar(&keyAlgo, "algo", "rsa",
-		"Public key algorithm, \"rsa\" or \"ecc\"")
-}
-
-func getAlgo() (tpm2.Algorithm, error) {
-	switch keyAlgo {
-	case "rsa":
-		return tpm2.AlgRSA, nil
-	case "ecc":
-		return tpm2.AlgECC, nil
-	case "":
-		panic("--algo flag not properly setup")
-	default:
-		return tpm2.AlgNull, fmt.Errorf("invalid argument %q for \"--algo\" flag", keyAlgo)
-	}
+	addPublicKeyAlgoFlag(pubkeyCmd)
```

Comment:
Good to use the `addPublicKeyAlgoFlag` helper functions here.

We should also use the `getEK` and `getAIK` helper methods in `getKey` below.

Reply:
I also added getEKwithAlgo and getSRKwithAlgo, for use cases like we need to provide the algorithm directly instead of referring from the Flag value. For example, when Unsealing, we need to get the algo from the sealed file instead of the Flag value.